substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -263,7 +288,20 @@ public void registerWithTargetConstructorClass(ConfigurationCondition condition,
             }
         }
 
+        String targetClassName = rawTargetClassName.contains("$$Lambda$") ? rawTargetClassName.split(LambdaUtils.SPLIT_BY_LAMBDA)[0] : rawTargetClassName;
         Class<?> serializationTargetClass = typeResolver.resolveType(targetClassName);
+
+        if (rawTargetClassName.contains("$$Lambda$") && !SerializationFeature.capturingClasses.contains(serializationTargetClass)) {
+            capturingClasses.add(serializationTargetClass);
+            RuntimeReflection.register(serializationTargetClass);
+            try {
+                RuntimeReflection.register(serializationTargetClass.getDeclaredMethod("$deserializeLambda$", SerializedLambda.class));
+            } catch (NoSuchMethodException e) {
+                VMError.shouldNotReachHere("Method named $deserializeLambda$ must exist in capturing class (" + serializationTargetClass.getName() + ") of lambdas that have writeReplace method.");
```

Comment:
in the capturing class ... the writeReplace method

Reply:
Fixed as suggested.